sonar-plugin-api/src/main/java/org/sonar/api/utils/Version.java

```diff
@@ -105,43 +124,58 @@ public static Version parse(String text) {
     if (!qualifier.isEmpty()) {
       s = substringBefore(s, QUALIFIER_SEPARATOR);
     }
-    List<String> split = SEQUENCE_SPLITTER.splitToList(s);
+    List<String> fields = SEQUENCE_SPLITTER.splitToList(s);
     int major = 0;
     int minor = 0;
-    int patch = 0;
-    int size = split.size();
+    int patch = DEFAULT_PATCH;
+    long buildNumber = DEFAULT_BUILD_NUMBER;
+    int size = fields.size();
     if (size > 0) {
-      major = parseSequence(split.get(0));
+      major = parseFieldAsInt(fields.get(0));
       if (size > 1) {
-        minor = parseSequence(split.get(1));
+        minor = parseFieldAsInt(fields.get(1));
         if (size > 2) {
-          patch = parseSequence(split.get(2));
+          patch = parseFieldAsInt(fields.get(2));
           if (size > 3) {
-            throw new IllegalArgumentException("Only 3 sequences are accepted");
+            buildNumber = parseFieldAsLong(fields.get(3));
+            if (size > 4) {
+              throw new IllegalArgumentException("Maximum 4 fields are accepted: " + text);
+            }
           }
         }
       }
     }
-    return new Version(major, minor, patch, qualifier);
+    return new Version(major, minor, patch, buildNumber, qualifier);
   }
 
   public static Version create(int major, int minor) {
-    return new Version(major, minor, 0, "");
+    return new Version(major, minor, DEFAULT_PATCH, DEFAULT_BUILD_NUMBER, DEFAULT_QUALIFIER);
   }
 
   public static Version create(int major, int minor, int patch) {
-    return new Version(major, minor, patch, "");
+    return new Version(major, minor, patch, DEFAULT_BUILD_NUMBER, DEFAULT_QUALIFIER);
   }
 
+  /**
+   * @deprecated in 6.3 to avoid ambiguity with buildId
+   */
+  @Deprecated
   public static Version create(int major, int minor, int patch, String qualifier) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)